deployer/src/main/java/eu/seaclouds/policy/SeaCloudsInitializerPolicy.java

```diff
@@ -0,0 +1,544 @@
+                    @Override
+                    public Void call(ConfigBag parameters) {
+                        LOG.info("Stopping SeaCloudsInitializerPolicy " + "for " + entity.getId());
+                        removeSlaAgreement();
+                        removeMonitoringRules();
+                        removeInfluxDbObservers();
+                        removeGrafanaDashboard();
+
+                        // Rewire the original behaviour
+                        ((StartableApplication) entity).stop();
+
+                        return null;
+                    }
+                })
+                .build();
+    }
+
+    private void installSlaAgreement() {
+        LOG.info("SeaCloudsInitializerPolicy is installing SLA Agreements for " + entity.getId());
+
+        HttpToolResponse response = post(getConfig(SLA_ENDPOINT) + "/seaclouds/agreements", MediaType.APPLICATION_XML_UTF_8.toString(),
+                getConfig(SLA_USERNAME), getConfig(SLA_PASSWORD), BaseEncoding.base64().decode(getConfig(SLA_AGREEMENT)));
+
+        if (!HttpTool.isStatusCodeHealthy(response.getResponseCode())) {
+            throw new RuntimeException("Something went wrong during the SLA Agreements installation. " +
+                    "Invalid response code, " + response.getResponseCode() + ":" + response.getContentAsString());
+        } else {
+            entity.sensors().set(SLA_ID, agreement.getAgreementId());
+        }
+
+    }
+
+    private void notifyRulesReady() {
+        LOG.info("SeaCloudsInitializerPolicy is starting to enforce SLA Agreements for " + entity.getId());
+
+        HttpToolResponse response = post(getConfig(SLA_ENDPOINT) + "/seaclouds/commands/rulesready?agreementId=" +
+                        agreement.getAgreementId(), MediaType.APPLICATION_XML_UTF_8.toString(),
+                getConfig(SLA_USERNAME), getConfig(SLA_PASSWORD), "".getBytes());
+
+        if (!HttpTool.isStatusCodeHealthy(response.getResponseCode())) {
+            throw new RuntimeException("Something went wrong during the SLA Agreements installation. " +
+                    "Invalid response code, " + response.getResponseCode() + ":" + response.getContentAsString());
+        }
+    }
+
+    private void installMonitoringRules() {
+        LOG.info("SeaCloudsInitializerPolicy is installing T4C Monitoring Rules for " + entity.getId());
+
+        HttpToolResponse response = post(getConfig(T4C_ENDPOINT) + "/v1/monitoring-rules", MediaType.APPLICATION_XML_UTF_8.toString(),
+                getConfig(T4C_USERNAME), getConfig(T4C_PASSWORD), BaseEncoding.base64().decode(getConfig(T4C_RULES)));
+
+        if (!HttpTool.isStatusCodeHealthy(response.getResponseCode())) {
+            throw new RuntimeException("Something went wrong while the monitoring rules installation. " +
+                    "Invalid response code, " + response.getResponseCode() + ":" + response.getContentAsString());
+        } else {
+            List<String> ruleIds = new ArrayList<>();
```

Comment:
again, why do you want to store this `T4C_IDS` in the application?
the method is called `installMonitoringRules` but this does more, no?